app/views/assets/publishing/batch_publishing_preview.html.erb

```diff
@@ -26,11 +31,34 @@
     <%= link_to "Back to profile", person_path(params[:id].to_i) -%>
   <% else %>
     <% @assets.sort_by { |k, v| v.first.class.name }.each do |type, items| %>
+      <div id="<%= items.first.class.table_name %>s">
       <h2><%= text_for_resource items.first %>(s)</h2>
+      <label><%= link_to "Select all #{(text_for_resource items.first).downcase}s", "#",
+                         id: "select_all_#{items.first.class.table_name}",
+                         onclick: "selectChildren(this,1); return false;" -%></label> |
+      <label><%= link_to "Deselect all #{(text_for_resource items.first).downcase}s", "#",
+                         id: "deselect_all_#{items.first.class.table_name}",
+                         onclick: "deselectChildren(this,1); return false;" -%></label>
       <% items.each do |item| %>
-        <%= render :partial => "assets/publishing/options_for_publishing", :object => item, :locals => {:html_classes => "publishing_options"} -%>
+        <%
+          case item
+          when Investigation
+            children = item.studies
+          when Study
+            children = item.assays
+          when Assay
+            children = item.assets
+          else
+            children = []
+          end
```

Comment:
I think instead of passing `children` to the options partial, it would be better to just pass a boolean to decide whether to show the "select all" buttons.

In this particular view I don't think it is necessary to show the "select all" buttons because it doesn't display the ISA items in a "tree" view, so the boolean could be set to false, and you can avoid having to hit the database for each ISA element to check if it has children.

Reply:
You are right, I overcomplicated the _options_for_publishing partial. Fixed in [ea1e959](https://github.com/seek4science/seek/pull/1399/commits/ea1e9598ceb4f8c0ccad8e3166308bfa0a3fdfc1).